lib/teslamate/log.ex

```diff
@@ -173,25 +173,37 @@ defmodule TeslaMate.Log do
   def get_positions_without_elevation(min_id \\ 0, opts \\ []) do
     limit = Keyword.get(opts, :limit, 100)
 
+    date_earliest =
+      cond do
+        min_id == 0 ->
+          DateTime.add(DateTime.utc_now(), -10, :day)
+
+        true ->
+          DateTime.from_iso8601("2003-07-01T00:00:00Z")
+      end
+
+    naive_date_earliest = DateTime.to_naive(date_earliest)
+
     non_streamed_drives =
       Repo.all(
-        from d in subquery(
-               from p in Position,
-                 select: %{
-                   drive_id: p.drive_id,
-                   streamed_count:
-                     count()
-                     |> filter(not is_nil(p.odometer) and is_nil(p.ideal_battery_range_km))
-                 },
-                 where: not is_nil(p.drive_id),
-                 group_by: p.drive_id
-             ),
-             select: d.drive_id,
-             where: d.streamed_count == 0
+        from p in Position,
+          select: p.drive_id,
+          inner_join: d in assoc(p, :drive),
+          where:
+            p.date > ^naive_date_earliest and
+              d.start_date > ^naive_date_earliest,
```

Comment:
How important is it to check the `start_date` here? If we could eliminate the check, we don't need to join the drive table, and that would presumably be more efficient... Would just checking `p.date` be sufficient?

Reply:
because if not adding p.date postgres is not making use of date index and has to remove lot of rows by filter (10x in my case) making query slow

without p.date

![grafik](https://github.com/user-attachments/assets/e1fffbc6-f906-4ed2-a2d4-158e4cb13ba6)

with p.date

![grafik](https://github.com/user-attachments/assets/8cee1328-e228-4b71-b948-778908a47991)

for the first query - yes, you are right. in that case d.start_date is sufficient as index positions_drive_id is used! will make an update!